consumers/sender/main.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	if exist {
+		return nil
+	}
+
+	if err := h.cache.Set(msg.UUID, m.Body, CacheDuration); err != nil {
+		logrus.WithFields(logrus.Fields{
+			"uuid":          msg.UUID,
+			"user_id":       msg.UserID,
+			"campaign_id":   msg.CampaignID,
+			"subscriber_id": msg.SubscriberID,
+		}).WithError(err).Error("Unable to write to cache")
+		return err
+	}
+
+	sendLog := &entities.SendLog{
+		UUID:         msg.UUID,
+		UserID:       msg.UserID,
+		CampaignID:   msg.CampaignID,
+		SubscriberID: msg.SubscriberID,
+		Status:       entities.StatusDone,
+	}
+
+	resp, err := sendEmail(*msg)
+	if err != nil {
+		sendLog.Status = entities.StatusFailed
+
+		// First check errors for retrying (returning) they don't need to be inserted in send logs
+		// also if the error is retryable delete it from cache
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ses.ErrCodeMessageRejected:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send bulk templated email. Message rejected.")
+			case ses.ErrCodeMailFromDomainNotVerifiedException:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send bulk templated email. Domain not verified.")
+			case ses.ErrCodeConfigurationSetDoesNotExistException:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send bulk templated email. Configuration set does not exist.")
+			default:
+				logrus.WithFields(logrus.Fields{
+					"uuid":          msg.UUID,
+					"user_id":       msg.UserID,
+					"campaign_id":   msg.CampaignID,
+					"subscriber_id": msg.SubscriberID,
+				}).WithError(aerr).Error("Unable to send templated email. Unknown status code.")
+			}
+		} else {
+			logrus.WithFields(logrus.Fields{
+				"uuid":          msg.UUID,
+				"user_id":       msg.UserID,
+				"campaign_id":   msg.CampaignID,
+				"subscriber_id": msg.SubscriberID,
+			}).WithError(err).Error("Unable to send templated email.")
+		}
+	}
+
+	if resp != nil {
+		sendLog.MessageID = resp.MessageId
+		sendLog.Description = resp.GoString()
+	}
+
+	err = h.storage.CreateSendLog(sendLog)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"uuid":             msg.UUID,
+			"user_id":          msg.UserID,
+			"campaign_id":      msg.CampaignID,
+			"subscriber_id":    msg.SubscriberID,
+			"send_bulk_status": resp.GoString(),
+		}).WithError(err).Error("Unable to add log for sent emails result.")
+	}
+
+	return nil
+}
+
+func sendEmail(msg entities.SenderTopicParams) (*ses.SendEmailOutput, error) {
```

Comment:
As a future idea, maybe this could be part of a service, and we can pass the SES client as a dependency so we can mock it for testing.